packages/core-data/src/utils/is-numeric-id.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * Checks a given string to determine if it's a numeric ID.
+ * For example, '123' is a numeric ID, but '123abc' is not.
+ *
+ * @param {string} str the string to determine if it's a numeric ID.
+ * @return {boolean} true if the string is a numeric ID, false otherwise.
+ */
+export default function isNumericID( str = '' ) {
```

Comment:
Do we need this function? What about `! isNaN( str )`

Reply:
That fails on the following test:

```
// should return false for input ""
expect( isNumericID( '' ) ).toBe( false );
// true
```

This can be tested in console:

```
! isNaN( '' )
// true
```

Reply:
Adding a small check for length would solve that I guess.

Reply:
I've updated to remove the default and added a test case for decimals.